### PR TITLE
Add new metrics

### LINKED ISSUE #337: Add new metrics
- https://github.com/tarantool/tarantool/commit/165d24f53622c2654995a729ba013e07c00e68a1 (vinyl: add regulator.blocked_writers stat)
- https://github.com/tarantool/tarantool/issues/5447
- https://github.com/tarantool/tarantool/pull/6429

### PR DESCRIPTION
**Pull request by yngvar-antonsson**
- tnt_replication_upstream_id_lag
- tnt_replication_downstream_id_lag
- tnt_vinyl_regulator_blocked_writers
- tnt_net_requests_in_progress_total
- tnt_net_requests_in_progress_current
- tnt_net_requests_in_stream_total
- tnt_net_requests_in_stream_current

I didn't forget about

- [x] Tests
- [x] Changelog
- [x] Documentation (README and rst)

Close #337

### CODE REVIEW

File: metrics/tarantool/info.lua [hunk 1]
```diff
@@ -23,9 +23,17 @@ local function update_info_metrics()
...
+            local metric_name_old = 'replication_' .. k .. '_lag'
+            collectors_list[metric_name_old] =
+                utils.set_gauge(metric_name_old, 'Replication lag for instance ' .. k, v.upstream.lag)
+            local metric_name = 'replication_upstream_' .. k .. '_lag'
             collectors_list[metric_name] =
-                utils.set_gauge(metric_name, 'Replication lag for instance ' .. k, v.upstream.lag)
+                utils.set_gauge(metric_name, 'Replication upstream lag for instance ' .. k, v.upstream.lag)
+        end
+        if v.downstream ~= nil then
+            local metric_name = 'replication_downstream' .. k .. '_lag'
```
Comment (olegrok) [inline metrics/tarantool/info.lua:34]: Seems that it should be a lablel. We shouldn't hardcode instance uuid to metric_name.

File: metrics/tarantool/vinyl.lua [hunk 2]
```diff
@@ -24,6 +24,10 @@ local function update()
     collectors_list.vinyl_regulator_dump_watermark =
         utils.set_gauge('vinyl_regulator_dump_watermark', 'Point when dumping must occur',
         vinyl_stat.regulator.dump_watermark)
+    collectors_list.vinyl_regulator_blocked_writers =
+        utils.set_gauge('vinyl_regulator_blocked_writers', 'The number of fibers that are blocked waiting ' ..
+            'for Vinyl level0 memory quota', vinyl_stat.regulator.blocked_writers)
```
Comment (olegrok) [inline metrics/tarantool/vinyl.lua:29]: Probably this metric will be available for versions 2.10+

File: CHANGELOG.md [hunk 3]
```diff
@@ -43,6 +50,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - `tnt_cpu_count`
   - `tnt_cpu_total`
   - `tnt_vinyl_scheduler_dump_count`
+  - `tnt_replication_<id>_lag`
```
[the remark this answers is not in the document]
Comment (yngvar-antonsson) [reply CHANGELOG.md:53]: Thank you!

File: metrics/init.lua [hunk 4]
```diff
@@ -126,7 +126,8 @@ return {
     enable_default_metrics = function(include, exclude)
         log.warn('metrics tnt_net_sent_rps, tnt_net_received_rps, tnt_net_connections_rps, '..
             'tnt_net_requests_rps, tnt_stats_op_rps, tnt_space_count, tnt_fiber_count, ' ..
-            'lj_gc_total, tnt_cpu_count, tnt_cpu_total, tnt_vinyl_scheduler_dump_count ' ..
+            'lj_gc_total, tnt_cpu_count, tnt_cpu_total, tnt_vinyl_scheduler_dump_count, ' ..
+            'tnt_replication_<id>_lag, tnt_replication_master/replica_<id>_lsn ' ..
```
Comment (olegrok) [inline metrics/init.lua:130]: We must not introduce deprecated metrics

Comment (yngvar-antonsson) [reply metrics/init.lua:130]: What do you mean? Those metrics are deprecated, and new ones are added to replace them.

Comment (olegrok) [reply metrics/init.lua:130]: Ouch
My bad, I incorrectly understood it.

File: test/utils.lua [hunk 5]
```diff
@@ -62,9 +62,14 @@ function utils.find_metric(metric_name, metrics_data)
     return #m > 0 and m or nil
 end
 
+
+local function to_number_multiple(...)
```
[the remark this answers is not in the document]
Comment (yngvar-antonsson) [reply test/utils.lua:66]: '8'<'10' == false

**Review by olegrok — APPROVED** [review]

File: test/utils.lua (hunk 5 above)
[the remark this answers is not in the document]
Comment (DifferentialOrange) [reply test/utils.lua:65]: Oh, I get it

File: test/tarantool/vinyl_test.lua [hunk 6]
```diff
@@ -26,5 +26,10 @@ g.test_vinyl_metrics_present = function()
     local metrics_cnt = fun.iter(metrics.collect()):filter(function(x)
         return x.metric_name:find('tnt_vinyl')
     end):length()
-    t.assert_equals(metrics_cnt, 20)
+    if utils.is_version_less(_TARANTOOL, '2.8.3')
+    and utils.is_version_greater(_TARANTOOL, '2.0.0') then
+        t.assert_equals(metrics_cnt, 20)
+    else
+        t.assert_equals(metrics_cnt, 21)
```
[the remark this answers is not in the document]
Comment (DifferentialOrange) [reply test/tarantool/vinyl_test.lua:33]: Oh, that's interesting. But I still think we should add strict version comparison here for 1.10

**Comment by DifferentialOrange** [comment]
Damn, I forgot to submit my review. If anything still seems relevant, please fix it in separate PR

**Pull request merged**